test/unit/math/src/math.cpp

```diff
@@ -15,8 +15,32 @@
 #include <alpaka/test/acc/TestAccs.hpp>
 #include <alpaka/test/queue/Queue.hpp>
 #include <alpaka/test/KernelExecutionFixture.hpp>
+
 #include <catch2/catch.hpp>
 
+using TestAccs = alpaka::test::acc::EnabledAccs<
+    alpaka::dim::DimInt< 1u >,
+    std::size_t
+>;
+
+using Functors =
+    alpaka::meta::Concatenate<
+        alpaka::test::unit::math::UnaryFunctors,
+        alpaka::test::unit::math::BinaryFunctors
+    >;
+
+using TestAccFunctorTuples =
+    alpaka::meta::CartesianProduct<
+        std::tuple,
+        TestAccs,
+        Functors
+    >;
```

Comment:
@jkrude Here is the meta programming that generates an entry per functor per accelerator.
You can do something similar for your use case e.g.:
```
using TestAccFunctorTuples =
    alpaka::meta::CartesianProduct<
        std::tuple,
        TestAccs,
        DataTypes
    >;
```
together with `TEMPLATE_LIST_TEST_CASE` and `std::tuple_element_t` you can iterate over and get the current types.

Reply:
Depending on the size of template lists you may get issues with the template instantiation depth in nvcc as I did here.